src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -531,13 +568,29 @@ public void addMapAnnotationToDataset(long datasetId, String key, String value)
 
 
 
-    } catch (Exception e) {
-      System.out.println(e);
+    } catch (DSOutOfServiceException dsOutOfServiceException) {
+      throw new RuntimeException("Error while accessing omero service: broken connection, expired session or not logged in", dsOutOfServiceException);
+    } catch (ExecutionException executionException) {
+      throw new RuntimeException("Task aborted unexpectedly.", executionException);
+    } catch (DSAccessException dsAccessException) {
+      throw new RuntimeException("Could not pull data from the omero server.", dsAccessException);
     }
 
     return imageInfo;
   }
 
+  public String imageViewerLink(long imageId) {
+    int serverId = 1;
```

Comment:
If this is comparable to the openBIS DSS id/name, maybe it should be in the config or at least a global variable.

Reply:
Yes, I am not sure how it is provided or if it is always `1`. Therefore I introduced the local field which we can change once we figured that out. Maybe @luiskuhn has some input here.

Reply:
I do not know whether we need access to it outside of this method. This is why it is a local variable atm.

Reply:
Okay, maybe at this stage it isn't necessary. But we made this mistake with openBIS and should fix it ASAP once we know more.
Maybe it is also possible to fetch the correct serverId from the imageID? That would be even better.

Reply:
> Maybe it is also possible to fetch the correct serverId from the imageID? That would be even better.

Tried that but found no such functionality. I can get the server version and more but not the server id.

Reply:
oh well, let's keep in in mind then :)